lib/config-generator.js

```diff
@@ -115,73 +78,112 @@ export class ConfigGenerator {
         const extensions = []; // file extensions to lint, the default is ["js", "mjs", "cjs"]
 
         let importContent = "";
-        const helperContent = `import path from "path";
-import { fileURLToPath } from "url";
-import { FlatCompat } from "@eslint/eslintrc";
-import pluginJs from "@eslint/js";
-
-// mimic CommonJS variables -- not needed if using CommonJS
-const __filename = fileURLToPath(import.meta.url);
-const __dirname = path.dirname(__filename);
-const compat = new FlatCompat({baseDirectory: __dirname, recommendedConfig: pluginJs.configs.recommended});
-`;
         let exportContent = "";
         let needCompatHelper = false;
+        const languages = this.answers.languages;
+        const purpose = this.answers.purpose;
 
-        if (this.answers.moduleType === "commonjs" || this.answers.moduleType === "script") {
-            exportContent += `  {files: ["**/*.js"], languageOptions: {sourceType: "${this.answers.moduleType}"}},\n`;
-        }
+        if (languages?.includes("javascript")) {
+            if (this.answers.moduleType === "commonjs" || this.answers.moduleType === "script") {
+                exportContent += `  {files: ["**/*.js"], languageOptions: {sourceType: "${this.answers.moduleType}"}},\n`;
+            }
 
-        if (this.answers.env?.length > 0) {
-            this.result.devDependencies.push("globals");
-            importContent += "import globals from \"globals\";\n";
-            const envContent = {
-                browser: "globals: globals.browser",
-                node: "globals: globals.node",
-                "browser,node": "globals: {...globals.browser, ...globals.node}"
-            };
+            if (this.answers.env?.length > 0) {
+                this.result.devDependencies.push("globals");
+                importContent += "import globals from \"globals\";\n";
+                const envContent = {
+                    browser: "globals: globals.browser",
+                    node: "globals: globals.node",
+                    "browser,node": "globals: {...globals.browser, ...globals.node}"
+                };
 
-            exportContent += `  {languageOptions: { ${envContent[this.answers.env.join(",")]} }},\n`;
-        }
+                exportContent += `  {languageOptions: { ${envContent[this.answers.env.join(",")]} }},\n`;
+            }
+
+            if (purpose === "syntax") {
+
+                // no need to install any plugin
+            } else if (purpose === "problems") {
```

Comment:
if we don't want to install in this case we can ignore it? and can only retain the other if?

Reply:
1. it implies that in this case we did it intentionally, rather than forgetting about the branch.
2. it's not part of this change, so let's try to focus more on the changed things :)